realm/realm-library/src/androidTest/java/io/realm/RealmTest.java

```diff
@@ -2321,4 +2322,97 @@ public void testIsEmpty() {
+                    testRealm.addChangeListener(this);
+                    allTypes[0] = testRealm.where(AllTypes.class).findFirst();
+                    allTypes[0].addChangeListener(dummyListener);
+
+                    testRealm.executeTransaction(new Realm.Transaction() {
+                        @Override
+                        public void execute(Realm realm) {
+                            realm.createObject(AllTypes.class);
+                        }
+                    }, new Realm.Transaction.Callback());
+                } else if (testRealm.where(AllTypes.class).count() == 2) {
+                    testRealm.removeChangeListener(this);
+                    testRealm.close();
+                    testRealm = null;
+
+                    testRealm = Realm.getInstance(testConfig);
+                    testRealm.addChangeListener(this);
+                    // Step 3: Test with results listener
+                    results[0] = testRealm.where(AllTypes.class).findAll();
+                    results[0].addChangeListener(dummyListener);
+
+                    testRealm.executeTransaction(new Realm.Transaction() {
+                        @Override
+                        public void execute(Realm realm) {
+                            realm.createObject(AllTypes.class);
+                        }
+                    }, new Realm.Transaction.Callback());
+                } else if (testRealm.where(AllTypes.class).count() == 3) {
+                    testRealm.removeChangeListener(this);
+                    testRealm.close();
+
+                    // Step 4: No other listeners
+                    testRealm = Realm.getInstance(testConfig);
+                    testRealm.addChangeListener(this);
+                    testRealm.executeTransaction(new Realm.Transaction() {
+                        @Override
+                        public void execute(Realm realm) {
+                            realm.createObject(AllTypes.class);
+                        }
+                    }, new Realm.Transaction.Callback());
+                } else if (testRealm.where(AllTypes.class).count() == 4) {
+                    Looper.myLooper().quit();
+                }
+            }
+        };
+
+        testRealm.addChangeListener(listener);
+
+        // Step 1: Empty change listener
+        allTypes[0] = testRealm.where(AllTypes.class).findFirstAsync();
+        allTypes[0].addChangeListener(dummyListener);
+
+        testRealm.executeTransaction(new Realm.Transaction() {
+            @Override
+            public void execute(Realm realm) {
+                realm.createObject(AllTypes.class);
+            }
+        }, new Realm.Transaction.Callback());
+
+        Handler handler = new Handler(Looper.myLooper());
+        handler.postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                // Timeout
+                assertTrue(false);
+            }
+        }, 4000);
```

Comment:
I guess, it is different from all other unit tests though. As a minimum 4000 should be a constant named TIMEOUT or similar.